### DISCUSSION TITLE
Missing boolean properties movable, scalable, rotatable

### DISCUSSION THREAD

**Discussion by abhimanusharma** [fabricjs/fabric.js#6976] [Q&A]
I have fabric.js latest version with an iText element to place on a specific coordinate, where I want to be able to just change/edit the text but not change its current position. Its position should be fixed and should not be stretched or skewed.

Here is my js code:
```
var text = new fabric.IText(textValue, {
    left: event.clientX - fabricObj.upperCanvasEl.getBoundingClientRect().left,
    top: event.clientY - fabricObj.upperCanvasEl.getBoundingClientRect().top,
    fill: inst.color,
    fontSize: inst.font_size,
    selectable: false
});
fabricObj.add(text);
```
Is there any property like movable which can be set to false or any other way?

There must be properties similar to selectable or eventable to make an object non-movable(only) or non-scalable(only)

I also tried to do it this way, 

```
text.on('moving', function() {
    text.set({eventable: false});
});
text.on('rotating', function() {
    text.set({eventable: false});
});
text.on('scaling', function() {
    text.set({eventable: false});
});
```

That's not working as well.

**Comment by asturur** [comment]
Hi, this is material for a discussion and not an issue.
There are docs for fabricJS here:

http://fabricjs.com/docs/fabric.Object.html

Look for lockMovement, lockScaling
For the rotation or for an alternative look at the custom control api
